src/core/platforms/android/androidplatformutilities.cpp

```diff
@@ -557,10 +559,25 @@ QVariantMap AndroidPlatformUtilities::sceneMargins( QQuickWindow *window ) const
   return margins;
 }
 
+void AndroidPlatformUtilities::uploadPendingAttachments( QFieldCloudConnection *connection ) const
+{
+  QTimer::singleShot( 500, [connection]() {
+    if ( connection )
```

Comment:
What is this check for exactly (this variable is not used after)

Reply:
It is for a/ trying to carry on the same logic across android and other platforms (i.e., if you don't pass a valid connection pointer, no upload takes place). It also essentially serves as a check to increase likelihood of having a valid user and token to do the upload too.